src/mantine-modals/src/ModalsProvider.tsx

```diff
@@ -87,12 +90,13 @@ export function ModalsProvider({ children, modalProps, labels, modals }: ModalsP
     return id;
   };
 
-  const closeModal = (id: string) => {
+  const closeModal = (id: string, canceled: boolean) => {
```

Comment:
Maybe this boolean param isn't the sexiest solution. I could have created another function called `cancelModal`, which would just include a lot of repeated logic inside.